File: Brocode/Python/7 - If Statements/main.py
# if = Do some block of statements only IF the condition is true
#      Else do something else (optional)

while True:
    try:
        age = int(input("Enter your age: "))
        break
    except ValueError as err:
        print("That is an invalid integer!")


if age >= 100:
    print("You are too old to sign up!")
elif age >= 18: 
    print("You are now signed up!")
elif age < 0:
    print("You have not been born yet!")
else:
    print("You are too young to sign up!")

name = input("Enter your name: ")
if name == "":
    print("No name!")
else:
    print(f"Hello {name}")


online = False

print("User is online") if online else print("User is offline")

# This is why switches are better when possible (literal elif structure):
# if ____ :
# else:
#   if:
#       else:
#           if:
#               else:

File: Brocode/Java/7 - If Statements/src/Main.java

public class Main {
	public static void main(String[] args) {
		
		// If statement => performs a block of code if
		//				   its condition is true
		//
		// 		- Operate on boolean variables
		//		- Can use various operators && || !
		//	    - Same syntax as C/C++
		// 		- Also have "one-liner" varients, but not
		//		  quite like those in Python/C# with the
		//	      no curly braces
		//      - Always be aware of the order that your 
		//		  if statements evaluate (sometimes areas can
		//		  be unreachable/not make sense)
		
		int age = 45;
		if (age >= 18) {
			System.out.println("You are an adult! ");
			System.out.println("You can drink in Quebec. ");
			if (age >= 19) {
				System.out.println("You can drink anywhere else. ");
				if (age >= 21) System.out.println("You can drink in a corrupt doodoo place");
			}
		}
		else if (age > 0) {
			System.out.println("You are a child");
		}
		else if (age == 0) {
			System.out.println("You are a newborn baby");
		}
		else {
			System.out.println("You haven't been born yet");
		}
	}
}
